tests/ops/op_math/test_exp.py

```diff
@@ -435,23 +439,33 @@ def test_decomposition_into_pauli_rot(self, base, base_string):
         pr = op.decomposition()[0]
         assert qml.equal(pr, qml.PauliRot(3.21, base_string, base.wires))
 
-    @pytest.mark.parametrize("op_class", has_unitary_generator_types)
-    def test_generator_decomposition(self, op_class):
+    @pytest.mark.parametrize("op_name", all_qubit_operators)
+    def test_generator_decomposition(self, op_name):
         """Check that Exp decomposes into a specific operator if ``base`` corresponds to the
         generator of that operator."""
+        op_class = getattr(qml.ops.qubit, op_name)
+
+        if not op_class.has_generator:
+            pytest.skip("Operator does not have a generator.")
+
         if op_class in {qml.DoubleExcitationMinus, qml.DoubleExcitationPlus}:
             pytest.skip("qml.equal doesn't work for `SparseHamiltonian` generators.")
 
         phi = 1.23
 
         wires = [0, 1, 2] if op_class.num_wires is AnyWires else list(range(op_class.num_wires))
 
+        # PauliRot and PCPhase each have an extra required arg
         if op_class is qml.PauliRot:
             op = op_class(phi, pauli_word="XYZ", wires=wires)
+        elif op_class is qml.PCPhase:
+            op = op_class(phi, dim=8, wires=wires)
         else:
             op = op_class(phi, wires=wires)
 
         exp = qml.evolve(op.generator(), coeff=-phi)
+        if not exp.has_decomposition:
+            pytest.skip("Operator does not have a decomposition")
```

Comment:
Why is this line necessary? I'd expect the exponential of a generator (times the parameter) to be equal to the operator in every scenario.

Reply:
Without this, if a suitable decomposition is not found by the loop, the test fails - there were some phase operators (`PCPhase`, `CPhase`, `PhaseShift` etc.) which were not previously included in the loop as possible decompositions or tested in the tests that triggered this error. I (probably mistakenly) assumed that they were not intended to be handled by the decomposition function, due to the error and their not being in the tests, so I also skipped them. To handle these, I might need a little guidance on how these operators should decompose here?

Reply:
See my suggestion below. Most of these test failures are for ops whose generator is a `qml.Projector`, which has `num_wires=AnyWires`. Thus `Exp.decomposition` previously skips most of the ops in `qml.ops.qubit`.

Reply:
Thanks - I'm skipping the `PCPhase` test and comparing the generator for `PhaseShift` and `U1` to both of these classes.